src/backend/expungeservice/expunger/analyze.py

```diff
@@ -118,3 +134,166 @@ def __init__(self, name, dob, cases):
+        check = 'Does the charge fall under statute: ' + str(statute)
+        # TODO implement this
+        return Check(check=check, result=True)
+
+    def _is_charge_in_statute_list(charge, statutes, desc):
+        check = ('Does the charge fall under any of these statutes: ' +
+                 ','.join(str(statute) for statute in statutes))
+        # TODO implement this
+        return Check(check=check, result=False, check_desc=desc)
+
+    def _is_charge_sex_crime(charge):
+        # TODO update
+        _statutes_sex_crimes = []
+        return RecordAnalyzer._is_charge_in_statute_list(
+            charge, _statutes_sex_crimes, 'Is the charge a sex crime')
+
+    def _is_charge_traffic_crime(charge):
+        # TODO update
+        _statutes_traffic_crimes = []
+        return RecordAnalyzer._is_charge_in_statute_list(
+            charge, _statutes_traffic_crimes, 'Is the charge a traffic crime')
+
+    def _have_open_case(self):
+        check = 'Is there a open case for the client'
+        result = any([case.state == CaseState.OPEN for case in self.client.cases])
+        return Check(check=check, result=result)
+
+    """
+    Run Time Eligibility analysis on the client and their records
+
+    TODO: make it return analysis for each charge as well (which is supposed
+    to be an update on eligibility date and relevan statutes)
+
+    Returns:
+        An Result instance
+    """
+    def time_eligibility(self):
+        checks = []
+
+        checks.append(self._have_open_case())
+        if checks[-1].result:
+            return Result(ResultCode.OPEN_CASE, checks)
+
+        # TODO implement the rest
+
+        return Result(ResultCode.NO_ACTION)
+
+    """
+    Run Type Eligibility analysis on a charge
+
+    Args:
+        charge: A Charge instance
+
+    Returns:
+        A Result instance
+    """
+    def type_eligibility(self, charge):
+        checks = []
+
+        checks.append(RecordAnalyzer._is_charge_level(charge, 'Felony', 'A'))
+        if checks[-1].result:
+            return ResultInElig_137_225_5(checks=checks)
+
+        checks.append(RecordAnalyzer._is_charge_sex_crime(charge))
+        if checks[-1].result:
+            return ResultInElig_137_225_5(checks=checks)
+
+        checks.append(RecordAnalyzer._is_charge_traffic_crime(charge))
+        if checks[-1].result:
+            return ResultInElig_137_225_5(checks=checks)
+
+        # TODO add remaining checks
+
+        return Result(ResultCode.FURTHER_ANALYSIS, checks)
+
+    """
+    Analyze which records are expungeable
+
+    The method sets the Charge instance's result attribute with type eligibility
+    analysis result.
+
+    Returns:
+        A tuple consisting of:
+            - A Result instance that describes the Time Eligibility analyis.
+    """
+    def analyze(self):
+        for case in self.client.cases:
+            for charge in case.charges:
+                charge.type_elig_result = self.type_eligibility(charge)
```

Comment:
Do you think it would be better to pass one charge to the expunger and evaluate that individually?, and then have a separate analyzer if necessary (dependent on whether or not we can use a db) to check if there are any open cases and the time eligibility for the group of charges?

Reply:
We can do that but I'm not sure if that's better. The analyze function assumes the caller would want both type and time-eligibility analysis for a client.

> have a separate analyzer if necessary (dependent on whether or not we can use a db)

How does that relate to the db?

Reply:
We could ask the db to return the most recent charge and decide time elig on that or is it more involved than just looking at the most recent charge?

Reply:
For time eligibility analysis, we would have to look at all the charges on the client. Time Eligibility of most recent charge also depends on other convictions in the past.

Reply:
OK, thanks for clarifying that.